pkg/segment/structs/segstructs.go

```diff
@@ -366,6 +367,37 @@ type AppendCmdOption struct {
 	Value      interface{}
 }
 
+type ToJsonExpr struct {
+	FieldsDtypes    []*ToJsonFieldsDtypeOptions
+	DefaultType     *ToJsonFieldsDtypeOptions
+	FillNull        bool
+	IncludeInternal bool
+	OutputField     string
+	AllFields       bool
+}
+
+type ToJsonCmdOption struct {
+	OptionType string
+	Value      any
+}
+
+type ToJsonFieldsDtypeOptions struct {
+	Dtype ToJsonDtypes
+	Regex *utils.GobbableRegex
+}
+
+type ToJsonDtypes uint8
+
+const (
+	TJ_None ToJsonDtypes = iota
+	TJ_Auto
+	TJ_Bool
+	TJ_Json
+	TJ_Num
+	TJ_Str
+	TJ_PP // post process
```

Comment:
nit: IMO the characters saved is not worth the extra cognitive cost of remembering what PP stands for
```suggestion
	TJ_PostProcess
```
I'm also not sure what post process means in this context, so it might be worth adding more info in the comment